ndless-sdk/libndls/show_msg_user_input.c

```diff
@@ -48,15 +50,14 @@ int show_msg_user_input(const char * title, const char * msg, char * defaultvalu
 	String request_struct[] = {s_msg, request_value};	
 
 	int no_error = _show_msgUserInput(0, request_struct, s_title->str, s_msg->str);
-	unsigned len_out = (no_error) ? request_value->len : 0;	
 	string_free(s_title);	
 	string_free(s_msg);
 
-	if(no_error && len_out > 0) {
+	if(no_error && request_value->len > 0) {
 		char * t = string_to_ascii(request_value);
-		*value_ref = malloc(sizeof(char)*(len_out+1));
```

Comment:
Good idea, I changed it.

On Tuesday, April 12, 2016 09:25:01 AM Fabian Vogt wrote:
> >  		char * t = string_to_ascii(request_value);
> > 
> > -		*value_ref = malloc(sizeof(char)*(len_out+1));
> 
> Why not just ```*value_ref = strdup(string_to_ascii(request_value))```?
> 
> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/ndless-nspire/Ndless/pull/32/files/1ac6d00dfa7b91c336fbd1
> ea856f81a6cec08539#r59406823